From e7877f9ea66fc9b0a1a9c4942a8db93f02326fd2 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Wed, 14 May 2014 10:16:36 +0200
Subject: [LIBREPORT PATCH] Ensure 'long long unsigned' in printf format args

The sizes of size_t are not equal on x86_64 and i686

Fixes a broken build on i686

Related to rhbz#1036585

Signed-off-by: Jakub Filak <jfilak@redhat.com>

Conflicts:

	tests/make_description.at
---
 tests/make_description.at | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/make_description.at b/tests/make_description.at
index dd9dcef..8e337e0 100644
--- a/tests/make_description.at
+++ b/tests/make_description.at
@@ -140,8 +140,8 @@ int main(int argc, char **argv)
             "Bugzilla",          14 - strlen("Bugzilla"),          "", "https://bugzilla.redhat.com/1000000",
             "RHTSupport",        14 - strlen("RHTSupport"),        "", "https://access.redhat.com/home",
             "ABRT Server",       14 - strlen("ABRT Server"),       "", "https://bug-report.itos.redhat.com",
-            FILENAME_BACKTRACE,  14 - strlen(FILENAME_BACKTRACE),  "", strlen(backtrace),
-            FILENAME_REPORTED_TO,14 - strlen(FILENAME_REPORTED_TO),"", strlen(get_problem_item_content_or_NULL(pd, FILENAME_REPORTED_TO))
+            FILENAME_BACKTRACE,  14 - strlen(FILENAME_BACKTRACE),  "", (long long unsigned)strlen(backtrace),
+            FILENAME_REPORTED_TO,14 - strlen(FILENAME_REPORTED_TO),"", (long long unsigned)strlen(get_problem_item_content_or_NULL(pd, FILENAME_REPORTED_TO))
             );
 
     if (strcmp(expected, description) != 0)
@@ -173,7 +173,7 @@ int main(int argc, char **argv)
             FILENAME_COUNT,      14 - strlen(FILENAME_COUNT),      "", "0",
             FILENAME_EXECUTABLE, 14 - strlen(FILENAME_EXECUTABLE), "", "/usr/bin/sh",
             FILENAME_PACKAGE,    14 - strlen(FILENAME_PACKAGE),    "", "libreport",
-            FILENAME_BACKTRACE,  14 - strlen(FILENAME_BACKTRACE),  "", strlen(backtrace)
+            FILENAME_BACKTRACE,  14 - strlen(FILENAME_BACKTRACE),  "", (long long unsigned)strlen(backtrace)
             );
 
     if (strcmp(expected, description) != 0)
-- 
1.8.3.1

